### PR TITLE
Moved all error strings to src/errors/messages and a few other things

### PR DESCRIPTION
**Pull request by SpaceEEC**
**Please describe the changes this PR makes and why it should be merged:**

- Moved all error strings to `src/errors/messages`
- Fixed / improved `Guild#setRolePosition` and `Guild#setChannelPosition`'s first arg validation
- Fixed a typo in `Guild#setChannelPosition` that throws an error when being run

Update
- Removed unused `PERMISSIONS_INVALID_FLAG` error message
- Fixed `PERMISSIONS_INVALID` that should be singular

**Semantic versioning classification:**  
- [ ] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/util/Snowflake.js [hunk 1]
```diff
@@ -9,7 +10,7 @@ let INCREMENT = 0;
  */
 class SnowflakeUtil {
   constructor() {
-    throw new Error(`The ${this.constructor.name} class may not be instantiated.`);
+    throw new Error('NOT_INSTANTIABLE', this.constructor.name);
```
Comment (devsnek) [inline src/util/Snowflake.js]: i purposely left these out because they aren't really "discord errors"

File: src/util/Collection.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
+const { Error, TypeError, RangeError } = require('../errors');
```
Comment (devsnek) [inline src/util/Collection.js]: collections shouldn't use discord errors

File: src/util/Util.js [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
+const { Error, RangeError, TypeError } = require('../errors');
 const has = (o, k) => Object.prototype.hasOwnProperty.call(o, k);
 
 /**
  * Contains various general-purpose utility methods. These functions are also available on the base `Discord` object.
  */
 class Util {
   constructor() {
-    throw new Error(`The ${this.constructor.name} class may not be instantiated.`);
+    throw new Error('NOT_INSTANTIABLE', this.constructor.name);
```
Comment (devsnek) [inline src/util/Util.js]: i purposely left these out because they aren't really "discord errors"

File: src/util/Collection.js (hunk 2 above)
Comment (bdistin) [inline src/util/Collection.js]: Oh missed this, I agree...

**Comment by SpaceEEC** [comment]
Followed requested changes

**Pull request merged**